### PR TITLE
[worklets] Fixes the setup of the inside setttings object for worklets.

### LINKED ISSUE #227: HTTPS state not defined for worklets
You say it comes from the global, but how did the global get it?

### PR DESCRIPTION
**Pull request by bfgeek**
Primarily based on:
https://html.spec.whatwg.org/#set-up-a-worker-environment-settings-object

Inherits a couple more things from the document.

Fixes #227 - inherits the HTTPS state from document

### CODE REVIEW

**Comment by bfgeek** [comment]
@annevk Can you take a look at this?

**Comment by bfgeek** [comment]
...also cc/ @domenic

File: worklets/Overview.bs [hunk 1]
```diff
@@ -265,24 +267,26 @@ When a user agent is to <dfn>set up a worklet environment settings object</dfn>,
...
 
         : The <a>API URL character encoding</a>
-        :: Return UTF-8.
+        :: Return <a>UTF-8</a>.
 
         : The <a>API base URL</a>
         :: Return |inheritedAPIBaseURL|.
 
-        : The <a>origin</a> and <a>effective script origin</a>
+        : The <a>origin</a>
```
Comment (annevk) [inline worklets/Overview.bs]: You should probably use for="environment settings object" for this one.

File: worklets/Overview.bs [hunk 2]
```diff
@@ -265,24 +267,26 @@ When a user agent is to <dfn>set up a worklet environment settings object</dfn>,
...
         :: Return |inheritedOrigin|.
```
Comment (annevk) [inline worklets/Overview.bs]: Is there any reason to be same-origin with your creator? Might be safer to return an opaque origin. Though I guess that should be considered separately.

File: worklets/Overview.bs (hunk 1 above)
Comment (bfgeek) [inline worklets/Overview.bs]: Done.

File: worklets/Overview.bs (hunk 2 above)
Comment (bfgeek) [inline worklets/Overview.bs]: Yeah I just had a quick read through the html spec where it was used. It seems sane to me to use an opaque origin. Changed. We can always switch back to inherited if needed.

**Pull request merged**